src/js/apps/patients/worklist/worklist_app.js

```diff
@@ -55,7 +55,7 @@ export default App.extend({
     this.setState(filterState.getFiltersState());
   },
   onChangeStateSort() {
-    if (!this.isRunning()) return;
+    if (!this.isRunning() || this.isRestarting()) return;
```

Comment:
I can't reproduce this bug in my dev environment, so can't tell if this actually fixes the issue.

Also still contemplating how to test this.

Reply:
Here's the test, but `isRestarting` doesn't fix the issue:
```js
    cy
      .intercept('GET', '/api/flows*', { delay: 1000, body: { data: [] } })
      .visit('/worklist/owned-by');

    cy
      .get('[data-date-filter-region]')
      .should('contain', 'Added:')
      .should('contain', 'This Month')
      .click();

    cy
      .get('.app-frame__pop-region')
      .contains('Last Week')
      .click();

    cy
      .get('.worklist-list__filter-sort')
      .click()
      .get('.picklist')
      .contains('Patient Last: A')
      .click();
```

Reply:
What about checking that the `list` view has a collection before setting the comparator?

Like:

```js
  onChangeStateSort() {
    if (!this.isRunning()) return;

    const listView = this.getChildView('list');

    if (listView.collection) {
      listView.setComparator(this.getComparator());
    }
  },
```